### PR TITLE
Update bgfg_gaussmix2.cpp to avoid divide by zero cases.

### PR DESCRIPTION
**Pull request by grundman**

### CODE REVIEW

File: modules/video/src/bgfg_gaussmix2.cpp [hunk 1]
```diff
@@ -693,11 +693,16 @@ class MOG2Invoker : public ParallelLoopBody
                 //go through all modes
                 //////
 
-                //renormalize weights
-                totalWeight = 1.f/totalWeight;
+                // Renormalize weights. In the special case that the pixel does
+                // not agree with any modes, set weights to zero (a new mode will be added below).
+                float invWeight = 0.f;
+                if (fabs(totalWeight) > 1e-12f) {
```
Comment (sovrasov) [inline modules/video/src/bgfg_gaussmix2.cpp]: I think `DBL_EPSILON` constant can be used here instead of `1e-12f`.

**Comment by grundman** [comment]
DBL_EPSILON change addressed

**Comment by grundman** [comment]
sorry - closed it on accident :)

**Comment by sovrasov** [comment]
:+1:

File: modules/video/src/bgfg_gaussmix2.cpp [hunk 2]
```diff
@@ -693,11 +693,16 @@ class MOG2Invoker : public ParallelLoopBody
                 //go through all modes
                 //////
 
-                //renormalize weights
-                totalWeight = 1.f/totalWeight;
+                // Renormalize weights. In the special case that the pixel does
+                // not agree with any modes, set weights to zero (a new mode will be added below).
+                float invWeight = 0.f;
+                if (fabs(totalWeight) > DBL_EPSILON) {
```
Comment (vpisarev) [inline modules/video/src/bgfg_gaussmix2.cpp]: could you use std::abs() here and comparison with FLT_EPSILON or such? We don't want to use double precision unless it's not absolutely necessary.

**Comment by grundman** [comment]
Addressed comments.

**Pull request merged**